plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlJdbcConnectionCreation.java

```diff
@@ -0,0 +1,181 @@
+                new TestCase("SELECT * FROM nation WHERE nationkey = 1", 5),
+                new TestCase("SELECT avg(nationkey) FROM nation", 4),
+                new TestCase("SELECT * FROM nation, region", 6),
+                new TestCase("SELECT * FROM nation n, region r WHERE n.regionkey = r.regionkey", 8),
+                new TestCase("SELECT * FROM nation JOIN region USING(regionkey)", 10),
+                new TestCase("SELECT * FROM information_schema.schemata", 1),
+                new TestCase("SELECT * FROM information_schema.tables", 1),
+                new TestCase("SELECT * FROM information_schema.columns", 247),
+                new TestCase("SELECT * FROM nation", 3),
+                new TestCase("CREATE TABLE copy_of_nation AS SELECT * FROM nation", 17),
+                new TestCase("INSERT INTO copy_of_nation SELECT * FROM nation", 15),
+                new TestCase("DELETE FROM copy_of_nation WHERE nationkey = 3", 4),
+                new TestCase("UPDATE copy_of_nation SET name = 'POLAND' WHERE nationkey = 1", 2, "This connector does not support updates"),
+                new TestCase("MERGE INTO copy_of_nation n USING region r ON r.regionkey= n.regionkey WHEN MATCHED THEN DELETE", 2, "This connector does not support merges"),
+                new TestCase("DROP TABLE copy_of_nation", 3),
+                new TestCase("SHOW SCHEMAS", 1),
+                new TestCase("SHOW TABLES", 2),
+                new TestCase("SHOW STATS FOR nation", 4));
+    }
+
+    private static DistributedQueryRunner createPostgreSqlQueryRunner(
+            TestingPostgreSqlServer server,
+            Iterable<TpchTable<?>> tables,
+            ConnectionCountingConnectionFactory connectionCountingConnectionFactory)
+            throws Exception
+    {
+        DistributedQueryRunner queryRunner = null;
+        try {
+            queryRunner = DistributedQueryRunner.builder(createSession())
+                    .build();
+
+            queryRunner.installPlugin(new TpchPlugin());
+            queryRunner.createCatalog("tpch", "tpch");
+
+            Map<String, String> connectorProperties = new HashMap<>();
+            connectorProperties.putIfAbsent("connection-url", server.getJdbcUrl());
+            connectorProperties.putIfAbsent("connection-user", server.getUser());
+            connectorProperties.putIfAbsent("connection-password", server.getPassword());
+            connectorProperties.putIfAbsent("postgresql.include-system-tables", "true");
+
+            queryRunner.installPlugin(new TestingPostgreSqlPlugin(connectionCountingConnectionFactory));
+            queryRunner.createCatalog("postgresql", "postgresql", connectorProperties);
+
+            copyTpchTables(queryRunner, "tpch", TINY_SCHEMA_NAME, createSession(), tables);
+
+            return queryRunner;
+        }
+        catch (Throwable e) {
+            closeAllSuppress(e, queryRunner, server);
```

Comment:
You don't need to close server, since it's `closeAfterClass`.

If you want to ensure it's closed faster, you'd need a similar try-catch block in the caller, since `server` is just an argument here, not allocated.